_rules/image-non-empty-accessible-name-23a2a8.md

```diff
@@ -228,10 +287,38 @@ This element is neither an `img` element nor has a role of `img`.
 <div aria-label="W3C logo"></div>
 ```
 
+#### Inapplicable Example 5
+
+This `img` element has no `src` attribute and no `alt` attribute so does not render an image and [represents nothing][].
+
+```html
+<img />
```

Comment:
Looking at this with fresh eyes. I don't think this is correct. These actually should fail SC 1.1.1. Here's my reasoning:

1. These img elements are content, because they are nodes in the accessibility tree
2. There is no text here, so if 1 is true, these quality as non-text content
3. Since these are invisible, the "Decoration, Formatting, Invisible" applies
4. Since these aren't marked up with `aria-hidden=true`,`role=none` or `alt=""`, assistive technologies will not consistently ignore them
5. Because these are not ignored by assistive technologies, the exception isn't met, and so 1.1.1 is not satisfied


I think I'd prefer to keep this rule as is to be honest. That ARIA in HTML explicitly says that an `img` with no `alt` attribute has a semantic role of `img` suggests to me this proposal not how the W3C expects this to be interpreted, even if some AT will ignore some of these (depending on the verbosity setting). It might be better just to mention this in the accessibility support section.

Reply:
Looking at the `<img/>` example (an image with no `src` and no `alt`). This isn't rendered on screen in Chrome/FF/Safari. It cannot render non-text content since it has no `src`.

It is exposed as `role=img` in Chrome/Firefox. The image does not appear in the accessibility tree in Safari.

There are two contradictory specs:

- html-map maps all `img` elements without `alt` to role=img (even if they have no src)
- the HTML spec says an `img` with no `alt and no `src` represents nothing (i.e. has no semantics)

The html-aam mapping is here:
https://www.w3.org/TR/html-aam-1.0/#el-img

The HTML spec says:

> If the [src](https://html.spec.whatwg.org/multipage/embedded-content.html#attr-img-src) attribute is not set and either the [alt](https://html.spec.whatwg.org/multipage/embedded-content.html#attr-img-alt) attribute is set to the empty string or the [alt](https://html.spec.whatwg.org/multipage/embedded-content.html#attr-img-alt) attribute is not set at all
The element [represents](https://html.spec.whatwg.org/multipage/dom.html#represents) nothing.

https://html.spec.whatwg.org/multipage/embedded-content.html#the-img-element

The HTML spec defines "represents nothing" as meaning the element has no semantics.

Note sure what do do here:

- you can make the argument `<img/>` **should not** fail 1.1.1 since there's no non-text content
- you can also make the argument `<img/>` **should** fail 1.1.1 since it's exposed as an role=img by html-aam

Reply:
I think we should raise the issue to the HTML AAM 🤔 
To see if `img` with no source should still have an image role or not.

Reply:
Issue filed at https://github.com/w3c/html-aam/issues/439